### PR TITLE
GEN-1683: consolidate Goerli tokens

### PR DESCRIPTION
**Pull request by fedealconada**
## Motivation

Re-deploying a bunch of Goerli contracts so all our factories/adapters use the same targets and underlyings and avoid having multiple ones for them.  

## Solution

- [x] redeploy Goerli contracts
- [x] created a task that redeploys Goerli contracts
- [x] created task that verifies adapters on Etherscan

NOTE: in the near future we may migrate these scripts to foundry   

## Implementer Checklist

- [ ]  [OPTIONAL] Create a reference implementation (in python or JS)
- [ ]  Create a short "Motivation" section for the PR
- [ ]  Write a spec for the feature and put it in the PR description – basic function names and expected state transitions is OK
- [ ]  Get the PR reviewed by at least two people

If this is your first time reviewing smart contract changes
- [ ]  Review the [Solcurity](https://github.com/Rari-Capital/solcurity) standard

If smart contract changes were made
- [ ]  Check all of the new revert paths with concrete tests
- [ ]  Check all of the new storage slot writes with concrete tests – do not make the passing test case the trivial case!
- [ ]  Go line-by-line through the new code and ensure it has all been covered by concrete tests – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
- [ ]  Simplify the implementation and spend some time trying to minimize gas costs
- [ ]  Write fuzz tests for the feature – try everything to break the implementation (is this function monotonically in/decreasing, should it always be less than something else, etc)
- [ ]  Capture bugs discovered during the above steps in concrete tests (regression tests)
- [ ]  Add integration tests – fuzz or concrete tests that test how the new code affects the entire system, either locally or with a mainnet fork
- [ ]  [OPTIONAL] Integrate the feature into the deployment scripts
- [ ]  [OPTIONAL] Add new sanity checks to the deployment scripts

If your PR uses or interacts with prices of any kind
- [ ] Make sure forge is running the test on the latest block
- [ ] Compare the value returned with some other source (like Curve)
- [ ] Check the `updatedAt` (if available) value of the contract and confirm that the price we are sourcing is in fact not older than X
- [ ] Ensure oracle liveness from the oracle maintainer, either through a public status interface or through written confirmation from a developer.

### CODE REVIEW

File: pkg/deployments/hardhat.utils.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ exports.STORAGE_SLOT = {
   USDC: 9,
   maDAI: 51,
   maUSDC: 51,
+  maUSDT: 51,
```
Comment (jparklev) [inline pkg/deployments/hardhat.utils.js:15]: probably you've already seen, but there's a nice way to find these with forge now `forge inspect --pretty AutoRoller storage`

File: pkg/deployments/hardhat.utils.js [hunk 2]
```diff
@@ -161,44 +162,45 @@ exports.delay = delay;
...
+        var m = JSON.parse(fs.readFileSync(`${path}/${file}`).toString());
+        delete m?.settings?.libraries[""]?.__CACHE_BREAKER__;
+        fs.writeFileSync(`${path}/${file}`, JSON.stringify(m));
+      });
 
-    console.log("Waiting 20 seconds for Etherscan to sync...");
-    await delay(20);
-    console.log("Trying to verify contract on Etherscan...");
-    try {
+      console.log("Waiting 10 seconds for Etherscan to sync...");
```
Comment (jparklev) [inline pkg/deployments/hardhat.utils.js:191]: Was this timing just trial and error?

File: pkg/deployments/tasks/verify-adapters/input.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      minm: ((365.25 * 24 * 60 * 60) / 12).toString(), // 1 month
+      maxm: (10 * 365.25 * 24 * 60 * 60).toString(), // 10 years
+      tilt: 0,
+      level: 31,
+      mode: 0, // 0 = monthly
+    },
+  },
+  {
+    contractName:
+      "@sense-finance/v1-core/src/adapters/abstract/erc4626/OwnableERC4626Adapter.sol:OwnableERC4626Adapter",
```
Comment (jparklev) [inline pkg/deployments/tasks/verify-adapters/input.js:58]: Shouldn't all the `OwnableERC4626Adapter` auto-verify once we get one?

File: pkg/deployments/tasks/goerli-tokens-factories/input.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    target: GOERLI_TOKENS.tokens[2].target,
+    maturities: [],
+  },
+];
+
+module.exports = {
+  5: {
+    divider: DIVIDER_1_2_0.get(CHAINS.GOERLI),
+    periphery: PERIPHERY_1_4_0.get(CHAINS.GOERLI),
+    oracle: "0xB3e70779c1d1f2637483A02f1446b211fe4183Fa",
```
Comment (jparklev) [inline pkg/deployments/tasks/goerli-tokens-factories/input.js]: Should this be stored as a constant as well?

File: pkg/deployments/tasks/goerli-tokens-factories/input.js [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    contractName: "MockOwnableAdapter",
+    ifee: ethers.utils.parseEther("0"), // 0%, no issuance fees
+    oracle: ethers.constants.AddressZero,
+    stake: WETH_TOKEN.get(CHAINS.GOERLI),
+    stakeSize: ethers.utils.parseEther("0.25"), // 0.25 WETH
+    minm: 0, // 0 days
+    maxm: (10 * 365.25 * 24 * 60 * 60).toString(), // 10 years
+    tilt: 0,
+    level: 31,
+    mode: 0, // 1 = weekly
```
Comment (jparklev) [inline pkg/deployments/tasks/goerli-tokens-factories/input.js]:
```suggestion
    mode: 1, // 1 = weekly
```
(same with the other adapters)

File: pkg/deployments/tasks/goerli-tokens-factories/index.js [hunk 6]
```diff
@@ -0,0 +1,547 @@
...
+
+  // Roll into the first Series
+  const autoRoller = await ethers.getContractAt("AutoRoller", autoRollerAddress, relayerSigner);
+  console.log(`\n- Roll Series on AutoRoller with OwnableAdapter whose target is ${t.name}`);
+  await autoRoller.callStatic.roll();
+  console.log(`  Successfully rolled Series!`);
+  console.log("\n\n");
+});
+
+// Deploys a new Space Factory (the one with getEqReserevs())
```
Comment (jparklev) [inline pkg/deployments/tasks/goerli-tokens-factories/index.js]:
```suggestion
// Deploys a new RLV Factory
```

**Review by jparklev — APPROVED** [review]
Gave a slightly softer review of the goerli stuff (which seems largely built off the `ownable-factory` task, right?), but I think it looks good. The goerli script runs OK on a hardhat goerli fork?

File: pkg/deployments/tasks/index.js [hunk 7]
```diff
@@ -18,3 +18,5 @@ require("./20221007-morpho-adapters");
 require("./20221031-factories-and-morpho");
 require("./20221112-periphery-v1.4");
 require("./20221114-ownable-factory");
+require("./goerli-tokens-factories");
```
Comment (jparklev) [inline pkg/deployments/tasks/index.js:21]: No date since this since we might use it repeatedly?

**Comment by jparklev** [comment]
> NOTE: in the near future we may migrate these scripts to foundry

⚒️ 🔥 


Also perhaps we give this PR checklist a "deployment code update" option with a different as compared to contract updates

File: pkg/deployments/tasks/goerli-tokens-factories/index.js [hunk 8]
```diff
@@ -0,0 +1,547 @@
...
+          `  * Sponsor Series for ${await adapterContract.symbol()} @ ${balancerVault} with maturity ${m}`,
+        );
+        await periphery.callStatic.sponsorSeries(adapterAddress, m, false);
+        await (await periphery.sponsorSeries(adapterAddress, m, false)).wait();
+      }
+    }
+
+    console.log("\n-------------------------------------------------------");
+  }
+});
```
Comment (abylaw) [inline pkg/deployments/tasks/goerli-tokens-factories/index.js:403]: could we break this large task into more subtasks/functions? It would be easier to understand that way :)

File: pkg/deployments/tasks/goerli-tokens-factories/index.js [hunk 9]
```diff
@@ -0,0 +1,547 @@
...
+
+  // create AutoRoller from OZ relayer
+  console.log(`\n- Deploy an AutoRoller for Ownable Adapter with target ${t.name} via AutoRollerFactory`);
+  const autoRollerFactory = await ethers.getContractAt("AutoRollerFactory", rlvFactory, relayerSigner);
+  let autoRollerAddress;
+  try {
+    autoRollerAddress = await autoRollerFactory.rollers(adapterAddress, 0);
+  } catch (e) {
+    console.log(`  ${adapterAddress} does not have an RLV. Creating one...`);
+  }
```
Comment (abylaw) [inline pkg/deployments/tasks/goerli-tokens-factories/index.js]: can use a promise to simplify:

```suggestion
  const autoRollerAddress = await autoRollerFactory.rollers(adapterAddress, 0)
      .catch(e => console.log(`  ${adapterAddress} does not have an RLV. Creating one...`));
```

File: pkg/deployments/tasks/goerli-tokens-factories/index.js [hunk 10]
```diff
@@ -0,0 +1,547 @@
...
+
+  if ((await stakeContract.allowance(OZ_RELAYER.get(CHAINS.GOERLI), periphery.address)).eq(0)) {
+    console.log(`\n- Approve Periphery to pull stake from relayer...`);
+    await stakeContract.approve(periphery.address, ethers.constants.MaxUint256).then(tx => tx.wait());
+  }
+
+  if ((await stakeContract.allowance(OZ_RELAYER.get(CHAINS.GOERLI), autoRollerAddress)).eq(0)) {
+    console.log(`  Approve AutoRoller to pull stake`);
+    await stakeContract.approve(autoRollerAddress, ethers.constants.MaxUint256).then(tx => tx.wait());
+  }
```
Comment (abylaw) [inline pkg/deployments/tasks/goerli-tokens-factories/index.js]: nit: pull into helper function?

**Review by abylaw — APPROVED** [review]
Nice! Just a few nits about maybe breaking things down into smaller function :b it gets a bit hard to follow after a while.

File: pkg/deployments/hardhat.utils.js (hunk 2 above)
Comment (fedealconada) [reply pkg/deployments/hardhat.utils.js:191]: Yeah haha, in any case, it seems like from now on we will be using forge scripts so probably we will be slowly deprecating all this.

File: pkg/deployments/tasks/verify-adapters/input.js (hunk 3 above)
Comment (fedealconada) [reply pkg/deployments/tasks/verify-adapters/input.js:58]: True but the very first one needs to be verified by us.

File: pkg/deployments/tasks/goerli-tokens-factories/input.js (hunk 5 above)
Comment (fedealconada) [reply pkg/deployments/tasks/goerli-tokens-factories/input.js]: yep, it's actually mode: 0, // 0 = monthly since this is an ownable adapter and must be 0

File: pkg/deployments/tasks/index.js (hunk 7 above)
Comment (fedealconada) [reply pkg/deployments/tasks/index.js:21]: right! that's the idea though probably next ones will be with forge scripts

**Comment by fedealconada** [comment]
> > NOTE: in the near future we may migrate these scripts to foundry
> 
> ⚒️ 🔥
> 
> Also perhaps we give this PR checklist a "deployment code update" option with a different as compared to contract updates

Not sure i got this hehe

File: pkg/deployments/tasks/goerli-tokens-factories/index.js (hunk 9 above)
Comment (fedealconada) [reply pkg/deployments/tasks/goerli-tokens-factories/index.js]: oh, yeah, you've also wrote this on other review! i need to get used to that!

**Comment by jparklev** [comment]
> > > NOTE: in the near future we may migrate these scripts to foundry
> > 
> > 
> > ⚒️ 🔥
> > Also perhaps we give this PR checklist a "deployment code update" option with a different as compared to contract updates
> 
> Not sure i got this hehe

Oh, just noticed all of the checklist items in the PR that don't apply to a change like this ~
